internal/gatewayapi/status/gateway.go

```diff
@@ -137,6 +137,13 @@ func computeGatewayProgrammedCondition(gw *gwapiv1.Gateway, deployment *appsv1.D
 			"No addresses have been assigned to the Gateway", time.Now(), gw.Generation)
 	}
 
+	if len(gw.Status.Addresses) > 16 {
+		return newCondition(string(gwapiv1.GatewayConditionProgrammed), metav1.ConditionFalse,
```

Comment:
thanks for improving the status message !
we should also discard `gw.Status.Addresses[16:len(gw.Status.Addresses)-1]` so the status update doesnt fail when writing into k8s API server